FreeRTOS-Plus/Source/Utilities/logging/logging_stack.h

```diff
@@ -37,25 +37,58 @@
 #include "logging_levels.h"
 
 /* Standard Include. */
+#include <stdio.h>
 #include <stdint.h>
 
+/* The macro definition for LIBRARY_LOG_NAME is for Doxygen
+ * documentation only. This macro is typically defined in only the
+ * <library>_config.h file or the demo_config.h file. */
+
+/**
+ * @brief The name of the library or demo to add as metadata in log messages
+ * from the library or demo.
+ *
+ * This metadata aids in identifying the module source of log messages.
+ * The metadata is logged in the format `[ <LIBRARY-NAME> ]` as a prefix to the
+ * log messages.
+ * Refer to #LOG_METADATA_FORMAT for the complete format of the metadata prefix in
+ * log messages.
+ */
+/* Check if LIBRARY_LOG_NAME macro has been defined. */
+#if !defined( LIBRARY_LOG_NAME )
+    #error "Please define LIBRARY_LOG_NAME for the library."
+#endif
+
 /* Metadata information to prepend to every log message. */
-#define LOG_METADATA_FORMAT    "[%s:%d] "
-#define LOG_METADATA_ARGS      __FUNCTION__, __LINE__
+#ifndef LOG_METADATA_FORMAT
```

Comment:
It looks like this is something users can define - if so it needs adding to the library documentation page.

Reply:
Yes, will add to the library documentation page